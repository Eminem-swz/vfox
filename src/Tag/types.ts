export type PatternType = 'light' | 'dark' | 'plain'
