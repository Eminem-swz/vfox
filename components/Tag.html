<!DOCTYPE html>
<html lang="zh-CN">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <title>Tag 标签 | Vfox for Vue3</title>
    <meta name="description" content="移动端 Vue 组件库">
    <link rel="stylesheet" href="/vfox/assets/style.82adce24.css">
    <link rel="modulepreload" href="/vfox/assets/Home.e3398b0e.js">
    <link rel="modulepreload" href="/vfox/assets/app.9b8f0faa.js">
    <link rel="modulepreload" href="/vfox/assets/components_Tag.md.cdee2a62.lean.js">
    <link rel="modulepreload" href="/vfox/assets/app.9b8f0faa.js">
    <meta name="twitter:title" content="Tag 标签 | Vfox for Vue3">
    <meta property="og:title" content="Tag 标签 | Vfox for Vue3">
  </head>
  <body>
    <div id="app"><!--[--><div class="theme"><header class="nav-bar" data-v-675d8756><div class="sidebar-button" data-v-675d8756><svg class="icon" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" role="img" viewBox="0 0 448 512"><path fill="currentColor" d="M436 124H12c-6.627 0-12-5.373-12-12V80c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12z" class></path></svg></div><a class="nav-bar-title" href="/vfox/" aria-label="Vfox for Vue3, back to home" data-v-675d8756 data-v-4a583abe><!----> Vfox for Vue3</a><div class="flex-grow" data-v-675d8756></div><div class="nav" data-v-675d8756><nav class="nav-links" data-v-675d8756 data-v-eab3edfe><!--[--><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/" data-v-b8818f8c>指南 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/design/" data-v-b8818f8c>设计 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item active" href="/vfox/components/Button" data-v-b8818f8c>组件 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/apis/Toast" data-v-b8818f8c>API <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-dropdown-link" data-v-eab3edfe data-v-56bf3a3f><button class="button" data-v-56bf3a3f><span class="button-text" data-v-56bf3a3f>选择版本</span><span class="right button-arrow" data-v-56bf3a3f></span></button><ul class="dialog" data-v-56bf3a3f><!--[--><li class="dialog-item" data-v-56bf3a3f><div class="nav-dropdown-link-item" data-v-56bf3a3f data-v-bbc27490><a class="item isExternal" href="https://cdn.fox2.cn/1.x/docs/" target="_blank" rel="noopener noreferrer" data-v-bbc27490><span class="arrow" data-v-bbc27490></span><span class="text" data-v-bbc27490>1.x</span><span class="icon" data-v-bbc27490><svg class="icon outbound" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewbox="0 0 100 100" width="15" height="15" data-v-bbc27490><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path><polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></span></a></div></li><li class="dialog-item" data-v-56bf3a3f><div class="nav-dropdown-link-item" data-v-56bf3a3f data-v-bbc27490><a class="item active" href="/vfox/" data-v-bbc27490><span class="arrow" data-v-bbc27490></span><span class="text" data-v-bbc27490>2.x</span><span class="icon" data-v-bbc27490><!----></span></a></div></li><!--]--></ul></div></div><!--]--><!----><!----></nav></div><!--[--><!--]--></header><aside class="sidebar" data-v-83e92a68><nav class="nav-links nav" data-v-83e92a68 data-v-eab3edfe><!--[--><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/" data-v-b8818f8c>指南 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/design/" data-v-b8818f8c>设计 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item active" href="/vfox/components/Button" data-v-b8818f8c>组件 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/apis/Toast" data-v-b8818f8c>API <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-dropdown-link" data-v-eab3edfe data-v-56bf3a3f><button class="button" data-v-56bf3a3f><span class="button-text" data-v-56bf3a3f>选择版本</span><span class="right button-arrow" data-v-56bf3a3f></span></button><ul class="dialog" data-v-56bf3a3f><!--[--><li class="dialog-item" data-v-56bf3a3f><div class="nav-dropdown-link-item" data-v-56bf3a3f data-v-bbc27490><a class="item isExternal" href="https://cdn.fox2.cn/1.x/docs/" target="_blank" rel="noopener noreferrer" data-v-bbc27490><span class="arrow" data-v-bbc27490></span><span class="text" data-v-bbc27490>1.x</span><span class="icon" data-v-bbc27490><svg class="icon outbound" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewbox="0 0 100 100" width="15" height="15" data-v-bbc27490><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path><polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></span></a></div></li><li class="dialog-item" data-v-56bf3a3f><div class="nav-dropdown-link-item" data-v-56bf3a3f data-v-bbc27490><a class="item active" href="/vfox/" data-v-bbc27490><span class="arrow" data-v-bbc27490></span><span class="text" data-v-bbc27490>2.x</span><span class="icon" data-v-bbc27490><!----></span></a></div></li><!--]--></ul></div></div><!--]--><!----><!----></nav><!--[--><!--]--><ul class="sidebar-links" data-v-83e92a68><!--[--><li class="sidebar-link"><p class="sidebar-link-item">基础</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Button">Button</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Icon">Icon</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Image">Image</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ScrollView">ScrollView</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Drawer">Drawer</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Modal">Modal</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Popover">Popover</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Dropdown">Dropdown</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">反馈</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Toast">Toast</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Dialog">Dialog</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Notify">Notify</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ActionSheet">ActionSheet</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/PopDialog">PopDialog</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/SwipeCell">SwipeCell</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">数据输入</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Form">Form</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Input">Input</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Radio">Radio</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Checkbox">Checkbox</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Slider">Slider</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Switch">Switch</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Stepper">Stepper</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Cascader">Cascader</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Picker">Picker</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Calendar">Calendar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/SearchBar">SearchBar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/NumberKeyboard">NumberKeyboard</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ImageUploader">ImageUploader</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/DatePicker">DatePicker</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Range">Range</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Rate">Rate</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">导航</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/BackTop">BackTop</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/NavBar">NavBar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/PopMenu">PopMenu</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Tab">Tab</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/SideTab">SideTab</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/TabBar">TabBar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/TabView">TabView</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Sticky">Sticky</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ScrollTab">ScrollTab</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/IndexView">IndexView</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Fixed">Fixed</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Pagination">Pagination</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">展示</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Layout">Layout</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Cell">Cell</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Group">Group</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Collapse">Collapse</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Empty">Empty</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Skeleton">Skeleton</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Swiper">Swiper</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ImagePreview">ImagePreview</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/FlatList">FlatList</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/NoticeBar">NoticeBar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Divider">Divider</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Badge">Badge</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Progress">Progress</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item active" href="/vfox/components/Tag">Tag</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#import">Import</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#props">Props</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#type-的合法值">type 的合法值</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#pattern-的合法值">pattern 的合法值</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#size-的合法值">size 的合法值</a><!----></li></ul></li><li class="sidebar-link"><a class="sidebar-link-item" href="#events">Events</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#slots">Slots</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#默认（-default）">默认（#default）</a><!----></li></ul></li></ul></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Order">Order</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/CountDown">CountDown</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Steps">Steps</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Price">Price</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Avatar">Avatar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Timeline">Timeline</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Stopwatch">Stopwatch</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Result">Result</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/LoadMore">LoadMore</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/TimeAgo">TimeAgo</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/CountUp">CountUp</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/CircleProgress">CircleProgress</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ActivityIndicator">ActivityIndicator</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">其他</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Copy">Copy</a><!----></li></ul></li><!--]--></ul><!--[--><!--]--></aside><div class="sidebar-mask"></div><main class="page" data-v-7eddb2c4><div class="container" data-v-7eddb2c4><!--[--><!--[--><!----><!--]--><!--[--><div><iframe style="width:375px;height:667px;margin-top:1.5rem;" src="https://godxiaoji.github.io/vfox/demo/#/Tag" frameborder="0"></iframe></div><!--]--><!--]--><div style="position:relative;" class="content" data-v-7eddb2c4><div><h1 id="tag-标签" tabindex="-1">Tag 标签 <a class="header-anchor" href="#tag-标签" aria-hidden="true">#</a></h1><h2 id="import" tabindex="-1">Import <a class="header-anchor" href="#import" aria-hidden="true">#</a></h2><div class="language-"><pre><code>import { Tag } from &#39;vfox&#39;
</code></pre></div><p>具体的引入方式可以参考<a href="./../#引入组件">引入组件</a>。</p><h2 id="props" tabindex="-1">Props <a class="header-anchor" href="#props" aria-hidden="true">#</a></h2><table><thead><tr><th>属性</th><th>类型</th><th>默认值</th><th>必填</th><th>说明</th></tr></thead><tbody><tr><td>type</td><td>string</td><td>&#39;default&#39;</td><td>否</td><td>标签类型</td></tr><tr><td>pattern</td><td>string</td><td>&#39;light&#39;</td><td>否</td><td>标签款式</td></tr><tr><td>color</td><td>string</td><td></td><td>否</td><td>自定义色彩，支持 hex rgb hsl 等写法，详细效果<a href="./../design/color.html#自定义色彩">查看</a></td></tr><tr><td>size</td><td>string</td><td>&#39;middle&#39;</td><td>否</td><td>标签尺寸</td></tr><tr><td>closable</td><td>boolean</td><td>true</td><td>否</td><td>是否展示关闭按钮，只推荐在 size=&#39;large&#39; 情况下开启</td></tr></tbody></table><h3 id="type-的合法值" tabindex="-1">type 的合法值 <a class="header-anchor" href="#type-的合法值" aria-hidden="true">#</a></h3><table><thead><tr><th>值</th><th>说明</th></tr></thead><tbody><tr><td>default</td><td>次要</td></tr><tr><td>primary</td><td>主要</td></tr><tr><td>success</td><td>成功</td></tr><tr><td>warning</td><td>警告</td></tr><tr><td>danger</td><td>危险</td></tr></tbody></table><h3 id="pattern-的合法值" tabindex="-1">pattern 的合法值 <a class="header-anchor" href="#pattern-的合法值" aria-hidden="true">#</a></h3><table><thead><tr><th>值</th><th>说明</th></tr></thead><tbody><tr><td>light</td><td>浅色标签，浅色底，深色字</td></tr><tr><td>dark</td><td>深色标签，深色底，白色字</td></tr><tr><td>plain</td><td>空心标签，白底，深色字，有边</td></tr></tbody></table><h3 id="size-的合法值" tabindex="-1">size 的合法值 <a class="header-anchor" href="#size-的合法值" aria-hidden="true">#</a></h3><table><thead><tr><th>值</th><th>说明</th></tr></thead><tbody><tr><td>middle</td><td>默认尺寸</td></tr><tr><td>small</td><td>小尺寸</td></tr></tbody></table><h2 id="events" tabindex="-1">Events <a class="header-anchor" href="#events" aria-hidden="true">#</a></h2><table><thead><tr><th>事件</th><th>描述</th><th>回调函数参数</th></tr></thead><tbody><tr><td>click</td><td>回到顶部按钮点击时触发</td><td>MouseEvent</td></tr><tr><td>close</td><td>关闭按钮点击时触发</td><td>{}</td></tr><tr><td>long-press</td><td>initialLongPress=true 长按标签时触发</td><td>{}</td></tr></tbody></table><h2 id="slots" tabindex="-1">Slots <a class="header-anchor" href="#slots" aria-hidden="true">#</a></h2><h3 id="默认（-default）" tabindex="-1">默认（#default） <a class="header-anchor" href="#默认（-default）" aria-hidden="true">#</a></h3><div class="language-"><pre><code>&lt;fx-tag&gt;标签&lt;/fx-tag&gt;
</code></pre></div></div></div><footer class="page-footer" data-v-7eddb2c4 data-v-fb8d84c6><div class="edit" data-v-fb8d84c6><div class="edit-link" data-v-fb8d84c6 data-v-1ed99556><!----></div></div><div class="updated" data-v-fb8d84c6><!----></div></footer><div class="next-and-prev-link" data-v-7eddb2c4 data-v-38ede35f><div class="container" data-v-38ede35f><div class="prev" data-v-38ede35f><a class="link" href="/vfox/components/Progress" data-v-38ede35f><svg xmlns="http://www.w3.org/2000/svg" viewbox="0 0 24 24" class="icon icon-prev" data-v-38ede35f><path d="M19,11H7.4l5.3-5.3c0.4-0.4,0.4-1,0-1.4s-1-0.4-1.4,0l-7,7c-0.1,0.1-0.2,0.2-0.2,0.3c-0.1,0.2-0.1,0.5,0,0.8c0.1,0.1,0.1,0.2,0.2,0.3l7,7c0.2,0.2,0.5,0.3,0.7,0.3s0.5-0.1,0.7-0.3c0.4-0.4,0.4-1,0-1.4L7.4,13H19c0.6,0,1-0.4,1-1S19.6,11,19,11z"></path></svg><span class="text" data-v-38ede35f>Progress</span></a></div><div class="next" data-v-38ede35f><a class="link" href="/vfox/components/Order" data-v-38ede35f><span class="text" data-v-38ede35f>Order</span><svg xmlns="http://www.w3.org/2000/svg" viewbox="0 0 24 24" class="icon icon-next" data-v-38ede35f><path d="M19.9,12.4c0.1-0.2,0.1-0.5,0-0.8c-0.1-0.1-0.1-0.2-0.2-0.3l-7-7c-0.4-0.4-1-0.4-1.4,0s-0.4,1,0,1.4l5.3,5.3H5c-0.6,0-1,0.4-1,1s0.4,1,1,1h11.6l-5.3,5.3c-0.4,0.4-0.4,1,0,1.4c0.2,0.2,0.5,0.3,0.7,0.3s0.5-0.1,0.7-0.3l7-7C19.8,12.6,19.9,12.5,19.9,12.4z"></path></svg></a></div></div></div><!--[--><!--]--></div></main></div><!----><!--]--></div>
    <script>__VP_HASH_MAP__ = JSON.parse("{\"apis_drawer.md\":\"6ae33ccb\",\"apis_modal.md\":\"86e51d05\",\"apis_notify.md\":\"935e74da\",\"apis_popover.md\":\"3d48f386\",\"apis_preview.md\":\"665ce18b\",\"apis_toast.md\":\"7cfd3b22\",\"apis_index.md\":\"adf4fdca\",\"components_actionsheet.md\":\"53586fe0\",\"components_activityindicator.md\":\"3cc4959c\",\"components_avatar.md\":\"a3d086f0\",\"components_backtop.md\":\"5c61eb80\",\"components_badge.md\":\"3b924acb\",\"components_button.md\":\"b6c67f33\",\"components_calendar.md\":\"41c222eb\",\"components_cascader.md\":\"71481780\",\"components_cell.md\":\"2bf2414e\",\"components_checkbox.md\":\"5f8d4ab0\",\"components_circleprogress.md\":\"1c9c07d3\",\"components_collapse.md\":\"42ac29f5\",\"components_copy.md\":\"6d236219\",\"components_countdown.md\":\"ad1444cf\",\"components_countup.md\":\"195e4fdd\",\"components_datepicker.md\":\"55685101\",\"components_dialog.md\":\"206e95e7\",\"components_divider.md\":\"1157c30e\",\"components_drawer.md\":\"bb7b75ca\",\"components_dropdown.md\":\"eafb06b5\",\"components_empty.md\":\"baf566f5\",\"components_fixed.md\":\"529aaf42\",\"components_flatlist.md\":\"c1860274\",\"components_form.md\":\"d1e77eb6\",\"components_group.md\":\"3245c718\",\"components_icon.md\":\"bc7951ff\",\"components_image.md\":\"ba8d9b5d\",\"components_imagepreview.md\":\"36f21909\",\"components_imageuploader.md\":\"dacbbf56\",\"components_indexview.md\":\"282a5ee4\",\"components_input.md\":\"40fe172b\",\"components_layout.md\":\"4e2af0d8\",\"components_loadmore.md\":\"f53fae48\",\"components_modal.md\":\"e2114f6a\",\"components_navbar.md\":\"1a61af44\",\"components_noticebar.md\":\"c24797dd\",\"components_notify.md\":\"e79a1dc6\",\"components_numberkeyboard.md\":\"a34b1044\",\"components_order.md\":\"6be336aa\",\"components_pagination.md\":\"7b5dbfc6\",\"components_picker.md\":\"d5f3ce45\",\"components_popdialog.md\":\"dd919f45\",\"components_popmenu.md\":\"63578017\",\"components_popover.md\":\"fd653d72\",\"components_price.md\":\"b26fc848\",\"components_progress.md\":\"c0498163\",\"components_radio.md\":\"2b359938\",\"components_range.md\":\"afc7429b\",\"components_rate.md\":\"dfa7583f\",\"components_result.md\":\"a28e5a44\",\"components_scrolltab.md\":\"221382e3\",\"components_scrollview.md\":\"61373341\",\"components_searchbar.md\":\"5874dc11\",\"components_sidetab.md\":\"068279c4\",\"components_skeleton.md\":\"71cec051\",\"components_slider.md\":\"54991480\",\"components_stepper.md\":\"edfeb91d\",\"components_steps.md\":\"ea6f12ab\",\"components_sticky.md\":\"faab9c9e\",\"components_stopwatch.md\":\"1069013c\",\"components_swipecell.md\":\"cce862d4\",\"components_swiper.md\":\"0652222d\",\"components_switch.md\":\"b1fd69df\",\"components_tab.md\":\"315469e5\",\"components_tabbar.md\":\"e908aed7\",\"components_tabview.md\":\"c4acdf46\",\"components_tag.md\":\"cdee2a62\",\"components_timeago.md\":\"4bdeccd5\",\"components_timeline.md\":\"48918486\",\"components_toast.md\":\"9c395cb2\",\"components_index.md\":\"cacdd688\",\"design_color.md\":\"704f5ddb\",\"design_index.md\":\"7a54c236\",\"index.md\":\"afef4054\"}")</script>
    <script type="module" async src="/vfox/assets/app.9b8f0faa.js"></script>
    
  </body>
</html>