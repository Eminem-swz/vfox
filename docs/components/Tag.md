# Tag 标签

## Import

```
import { Tag } from 'vfox'
```

具体的引入方式可以参考[引入组件](../index.md#引入组件)。

## Props

| 属性     | 类型    | 默认值    | 必填 | 说明                                                                               |
| -------- | ------- | --------- | ---- | ---------------------------------------------------------------------------------- |
| type     | string  | 'default' | 否   | 标签类型                                                                           |
| pattern  | string  | 'light'   | 否   | 标签款式                                                                           |
| color    | string  |           | 否   | 自定义色彩，支持 hex rgb hsl 等写法，详细效果[查看](../design/color.md#自定义色彩) |
| size     | string  | 'middle'  | 否   | 标签尺寸                                                                           |
| closable | boolean | true      | 否   | 是否展示关闭按钮，只推荐在 size='large' 情况下开启                                 |

### type 的合法值

| 值      | 说明 |
| ------- | ---- |
| default | 次要 |
| primary | 主要 |
| success | 成功 |
| warning | 警告 |
| danger  | 危险 |

### pattern 的合法值

| 值    | 说明                         |
| ----- | ---------------------------- |
| light | 浅色标签，浅色底，深色字     |
| dark  | 深色标签，深色底，白色字     |
| plain | 空心标签，白底，深色字，有边 |

### size 的合法值

| 值     | 说明     |
| ------ | -------- |
| middle | 默认尺寸 |
| small  | 小尺寸   |

## Events

| 事件       | 描述                                 | 回调函数参数 |
| ---------- | ------------------------------------ | ------------ |
| click      | 回到顶部按钮点击时触发               | MouseEvent   |
| close      | 关闭按钮点击时触发                   | {}           |
| long-press | initialLongPress=true 长按标签时触发 | {}           |

## Slots

### 默认（#default）

```
<fx-tag>标签</fx-tag>
```
