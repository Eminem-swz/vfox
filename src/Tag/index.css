.fx-tag {
  --tag-icon-size: 16px;
  --tag-color: #fff;
  --tag-background-color: #f5f5f5;
  --tag-border-color: #d9d9d9;
  display: inline-flex;
  align-items: center;
  height: 24px;
  padding: 0 12px;
  border-radius: 2px;
  font-size: 12px;
  color: var(--tag-color);
  background-color: var(--tag-background-color);
  border-style: solid;
  border-width: 0;
  border-color: var(--tag-border-color);
  box-sizing: border-box;
  -webkit-user-select: none;
  user-select: none;
}

.fx-tag + .fx-tag {
  margin-left: 12px;
}

.fx-tag .fx-icon {
  --size: var(--tag-icon-size);
  --color: var(--tag-color);
  cursor: pointer;
  margin: 0 -2px 0 2px;
}

.fx-tag .fx-icon:active {
  border-radius: 2px;
  background-color: rgba(0, 0, 0, 0.16);
}

.fx-tag.size--large .fx-icon {
  margin: 0 -4px 0 4px;
}

.fx-tag.size--large.pattern--plain {
  padding: 0 11px;
  border-width: 1px;
}

.fx-tag.size--middle {
  height: 18px;
  font-size: 10px;
  padding: 0 8px;
  --tag-icon-size: 12px;
}

.fx-tag.size--middle.pattern--plain {
  padding: 0 7px;
  border-width: 1px;
}

.fx-tag.size--small {
  height: 14px;
  font-size: 9px;
  padding: 0 4px;
  --tag-icon-size: 10px;
}

.fx-tag.size--small.pattern--plain {
  padding: 0 3px;
  border-width: 1px;
}

.fx-tag.type--default {
  --tag-color: #8c8c8c;
}

.fx-tag.type--default.pattern--dark {
  --tag-background-color: #8c8c8c;
  --tag-color: #fff;
}

.fx-tag.type--default.pattern--plain {
  --tag-background-color: #fff;
  --tag-border-color: #d9d9d9;
}

.fx-tag.type--primary {
  --tag-color: #1890ff;
  --tag-background-color: #e6f7ff;
}

.fx-tag.type--primary.pattern--dark {
  --tag-background-color: #1890ff;
  --tag-color: #fff;
}

.fx-tag.type--primary.pattern--plain {
  --tag-background-color: #fff;
  --tag-border-color: #40a9ff;
}

.fx-tag.type--success {
  --tag-color: #52c41a;
  --tag-background-color: #f0ffe0;
}

.fx-tag.type--success.pattern--dark {
  --tag-background-color: #52c41a;
  --tag-color: #fff;
}

.fx-tag.type--success.pattern--plain {
  --tag-background-color: #fff;
  --tag-border-color: #73d13d;
}

.fx-tag.type--warning {
  --tag-color: #faad14;
  --tag-background-color: #fffae0;
}

.fx-tag.type--warning.pattern--dark {
  --tag-background-color: #faad14;
  --tag-color: #fff;
}

.fx-tag.type--warning.pattern--plain {
  --tag-background-color: #fff;
  --tag-border-color: #ffc53d;
}

.fx-tag.type--danger {
  --tag-color: #ff4d4f;
  --tag-background-color: #ffe7e6;
}

.fx-tag.type--danger.pattern--dark {
  --tag-background-color: #ff4d4f;
  --tag-color: #fff;
}

.fx-tag.type--danger.pattern--plain {
  --tag-background-color: #fff;
  --tag-border-color: #ff7875;
}
