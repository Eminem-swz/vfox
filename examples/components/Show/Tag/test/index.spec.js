/* global describe,test,expect */

import { shallowMount } from '@vue/test-utils'
import Tag from '@/Tag'

const defaultSlot = 'tag'

describe('Tag', () => {
  test('should render default slot correctly', () => {
    const wrapper = shallowMount(Tag, {
      slots: {
        default: defaultSlot
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('should render prop closable=true correctly', () => {
    const wrapper = shallowMount(Tag, {
      propsData: {
        closable: true
      },
      slots: {
        default: defaultSlot
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
