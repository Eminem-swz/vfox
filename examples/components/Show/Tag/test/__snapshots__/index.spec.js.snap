// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag should render default slot correctly 1`] = `
<div class="fx-tag type--default size--large pattern--light">tag
  <!---->
</div>
`;

exports[`Tag should render prop closable=true correctly 1`] = `
<div class="fx-tag type--default size--large pattern--light">tag <icon-stub icon="CloseOutlined" size="0" width="0" height="0" class=""></icon-stub>
</div>
`;
